### PR TITLE
MP-8767 - Refactor DateTimeRange to use DatetimeOffset in SqlServer and Datetime (utc) in Postgres

### PR DESCRIPTION
**Pull request by danield-ap**

### CODE REVIEW

**Comment by danield-ap** [comment]
Resolves NoxOrg/Nox.Generator#579

File: src/Nox.EntityFramework.Postgres/Types/DateTimeRange/PostgresDateTimeRangeConverter.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿using Microsoft.EntityFrameworkCore.Storage.ValueConversion;
```
Comment (rochar) [inline src/Nox.EntityFramework.Postgres/Types/DateTimeRange/PostgresDateTimeRangeConverter.cs]: this is the same as PostgresDateTimeConverter, so should not be needed

File: src/Nox.EntityFramework.Postgres/Types/DateTimeRange/PostgresDateTimeRangeDatabaseConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        NoxSolutionCodeGeneratorState noxSolutionCodeGeneratorState,
+        EntityTypeBuilder builder,
+        NoxSimpleTypeDefinition property,
+        Entity entity, bool isKey)
+    {
+        builder
+            .OwnsOne(typeof(Nox.Types.DateTimeRange), property.Name, dtr =>
+            {
+                dtr.Property(nameof(Nox.Types.DateTimeRange.Start))
+                    .HasConversion<PostgresDateTimeRangeConverter>();
```
Comment (rochar) [inline src/Nox.EntityFramework.Postgres/Types/DateTimeRange/PostgresDateTimeRangeDatabaseConfiguration.cs]:
```suggestion
                    .HasConversion<PostgresDateTimeConverter>();
```

File: src/Nox.EntityFramework.Postgres/Types/DateTimeRange/PostgresDateTimeRangeDatabaseConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+                dtr.Property(nameof(Nox.Types.DateTimeRange.End))
+                    .HasConversion<PostgresDateTimeRangeConverter>();
```
Comment (rochar) [inline src/Nox.EntityFramework.Postgres/Types/DateTimeRange/PostgresDateTimeRangeDatabaseConfiguration.cs]:
```suggestion
                    .HasConversion<PostgresDateTimeConverter>();
```

**Review by rochar — APPROVED** [review]
sounds good, just fix the comments before merging

**Review by gjorgji-grgovski-authoritypartners — APPROVED** [review]

File: src/Nox.EntityFramework.Postgres/Types/DateTimeRange/PostgresDateTimeRangeDatabaseConfiguration.cs (hunk 2 above)
Comment (danield-ap) [reply src/Nox.EntityFramework.Postgres/Types/DateTimeRange/PostgresDateTimeRangeDatabaseConfiguration.cs]: `PostgresDateTimeRangeConverter` deleted.

File: src/Nox.EntityFramework.Postgres/Types/DateTimeRange/PostgresDateTimeRangeDatabaseConfiguration.cs (hunk 3 above)
Comment (danield-ap) [reply src/Nox.EntityFramework.Postgres/Types/DateTimeRange/PostgresDateTimeRangeDatabaseConfiguration.cs]: `PostgresDateTimeRangeDatabaseConfiguration` updated.

**Pull request merged**